### PR TITLE
Add the docs for temporary tables

### PR DESCRIPTION
**Pull request by djshow832**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by tiancaiamao — APPROVED** [review]

**Review by lcwangchao — APPROVED** [review]

File: temporary-table.md [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+
+```sql
+SELECT * FROM users;
+```
+
+```
+Empty set (0.00 sec)
+```
+
+注意，如果事务是自动提交的，插入的数据在 SQL 结束后被自动清空，导致后续 SQL 查找不到结果。因此应该使用非自动提交的事务读写全局临时表。
```
Comment (TomShawn) [inline temporary-table.md:252]:
```suggestion
> **注意：**
>
> 如果事务是自动提交的，插入的数据在 SQL 语句执行结束后会被自动清空，导致后续 SQL 执行查找不到结果。因此应该使用非自动提交的事务读写全局临时表。
```

File: temporary-table.md [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+## 临时表的约束和限制
+
+### 内存限制
+
+无论定义表时声明的 `EGNINE` 是哪种存储引擎，本地临时表和全局临时表的数据都只暂存在 TiDB 实例的内存中，不持久化。
+
+为了避免 OOM，用户可通过系统变量 [`tidb_tmp_table_max_size`](/system-variables.md#tidb_tmp_table_max_size-从-v53-版本开始引入) 限制每张临时表的大小，当临时表大小超过限制后报错。`tidb_tmp_table_max_size` 的默认值是 `64MB`。
```
Comment (TomShawn) [inline temporary-table.md:260]:
```suggestion
为了避免内存溢出，用户可通过系统变量 [`tidb_tmp_table_max_size`](/system-variables.md#tidb_tmp_table_max_size-从-v53-版本开始引入) 限制每张临时表的大小。当临时表大小超过限制后 TiDB 会报错。`tidb_tmp_table_max_size` 的默认值是 `64MB`。
```

File: temporary-table.md [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+```
+Empty set (0.00 sec)
+```
+
+注意，如果事务是自动提交的，插入的数据在 SQL 结束后被自动清空，导致后续 SQL 查找不到结果。因此应该使用非自动提交的事务读写全局临时表。
+
+## 临时表的约束和限制
+
+### 内存限制
```
Comment (TomShawn) [inline temporary-table.md:256]:
```suggestion
## 限制临时表的内存占用
```

File: temporary-table.md [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+无论定义表时声明的 `EGNINE` 是哪种存储引擎，本地临时表和全局临时表的数据都只暂存在 TiDB 实例的内存中，不持久化。
+
+为了避免 OOM，用户可通过系统变量 [`tidb_tmp_table_max_size`](/system-variables.md#tidb_tmp_table_max_size-从-v53-版本开始引入) 限制每张临时表的大小，当临时表大小超过限制后报错。`tidb_tmp_table_max_size` 的默认值是 `64MB`。
+
+例如调整 `tidb_tmp_table_max_size` 为 `256MB`：
```
Comment (TomShawn) [inline temporary-table.md:262]:
```suggestion
例如，将每张临时表的大小限制为 `256MB`：
```

File: temporary-table.md [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+```
+
+```
+Empty set (0.00 sec)
+```
+
+注意，如果事务是自动提交的，插入的数据在 SQL 结束后被自动清空，导致后续 SQL 查找不到结果。因此应该使用非自动提交的事务读写全局临时表。
+
+## 临时表的约束和限制
+
```
Comment (TomShawn) [inline temporary-table.md:255]:
```suggestion
```

File: temporary-table.md [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+例如调整 `tidb_tmp_table_max_size` 为 `256MB`：
+
+{{< copyable "sql" >}}
+
+```sql
+SET GLOBAL tidb_tmp_table_max_size=268435456;
+```
+
+### 与其他功能的兼容性
```
Comment (TomShawn) [inline temporary-table.md:270]:
```suggestion
### 与其他 TiDB 功能的兼容性限制
```

File: temporary-table.md [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+
+以下是本地临时表和全局临时表都不支持的功能：
+
+- 不支持 `AUTO_RANDOM` 列
+- 不支持 `SHARD_ROW_ID_BITS` 和 `PRE_SPLIT_REGIONS` 表选项
+- 不支持 partition table
```
Comment (TomShawn) [inline temporary-table.md:276]:
```suggestion
- 不支持分区表
```

File: temporary-table.md [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+- 不支持 `SPLIT REGION` 语句
+- 不支持 `ADMIN CHECK TABLE` 和 `ADMIN CHECKSUM TABLE` 语句
+- 不支持 `FLASHBACK TABLE` 和 `RECOVER TABLE` 语句
+- 不支持以临时表为源表执行 `CREATE TABLE LIKE` 语句
+- 不支持 stale read
+- 不支持 foreign key
```
Comment (TomShawn) [inline temporary-table.md:282]:
```suggestion
- 不支持 Stale Read
- 不支持外键
```

File: temporary-table.md [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+- 不支持 SQL binding
+- 不支持添加 TiFlash 副本
+- 不支持在临时表上创建视图
+- 不支持 placement rules
```
Comment (TomShawn) [inline temporary-table.md:286]:
```suggestion
- 不支持 Placement Rules
```

File: temporary-table.md [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+- 包含临时表的执行计划不会被 prepared plan cache 缓存
+
+以下是只有本地临时表不支持的功能：
+
+- 不支持通过系统变量 `tidb_snapshot` 读取历史数据
+
+## 生态工具支持
```
Comment (TomShawn) [inline temporary-table.md:293]:
```suggestion
## TiDB 生态工具支持
```

File: temporary-table.md [hunk 11]
```diff
@@ -0,0 +1,309 @@
...
+
+本地临时表只对当前会话可见，因此本地临时表不会被生态工具导出、备份、同步。
```
Comment (TomShawn) [inline temporary-table.md:295]:
```suggestion
本地临时表只对当前会话可见，因此本地临时表不会被 TiDB 生态工具导出、备份、同步。
```

File: temporary-table.md [hunk 12]
```diff
@@ -0,0 +1,309 @@
...
+
+全局临时表的表定义全局可见，因此全局临时表的表定义会被生态工具导出、备份、同步，但不导出数据。
```
Comment (TomShawn) [inline temporary-table.md:297]:
```suggestion
全局临时表的表定义全局可见，因此全局临时表的表定义会被 TiDB 生态工具导出、备份、同步，但不导出数据。
```

File: temporary-table.md [hunk 13]
```diff
@@ -0,0 +1,319 @@
+---
+title: 临时表
+---
+
+# 临时表
+
+本文介绍 TiDB 的临时表。
+
```
Comment (TomShawn) [inline temporary-table.md:8]:
```suggestion
```

File: temporary-table.md [hunk 14]
```diff
@@ -0,0 +1,319 @@
...
+TiDB 在 v5.3.0 版本中引入了临时表功能。该功能针对业务中间计算结果的临时存储问题，让用户免于频繁地建表和删表等操作。用户可将业务上的中间计算数据存入临时表，用完数据后 TiDB 自动清理回收临时表。这避免了用户业务过于复杂，减少了表管理开销，并提升了性能。
+
```
Comment (TomShawn) [inline temporary-table.md:10]:
```suggestion

本文介绍了 TiDB 临时表的使用场景、临时表类型、使用示例、限制临时表内存占用的方法、与其他 TiDB 功能的兼容性限制。

```

File: temporary-table.md [hunk 15]
```diff
@@ -0,0 +1,319 @@
+---
+title: 临时表
```
Comment (TomShawn) [inline temporary-table.md:2]:
```suggestion
title: 临时表
summary: 使用 TiDB 中的临时表功能存储业务的中间数据。
```

**Review by TomShawn — APPROVED** [review]
Rest LGTM

File: temporary-table.md [hunk 16]
```diff
@@ -0,0 +1,322 @@
...
+- 本地临时表的数据对会话内的所有事务可见
+- 在会话结束后，该会话创建的本地临时表会被自动删除
+- 本地临时表可以与普通表同名，此时在 DDL 和 DML 语句中，普通表被隐藏，直到本地临时表被删除
+
+用户可通过 `CREATE TEMPORARY TABLE` 语句创建本地临时表，通过 `DROP TABLE` 或 `DROP TEMPORARY TABLE` 语句删除本地临时表。
+
+不同于 MySQL，TiDB 本地临时表都是外部表，SQL 语句不会创建内部临时表。
+
+### 本地临时表使用示例
+
```
Comment (TomShawn) [inline temporary-table.md:43]:
```suggestion

> **注意：**
>
> 使用 TiDB 中的临时表前，注意临时表[与其他 TiDB 功能的兼容性限制](#与其他 TiDB 功能的兼容性限制)。

```

File: temporary-table.md [hunk 17]
```diff
@@ -0,0 +1,322 @@
...
+全局临时表是 TiDB 的扩展功能，它有以下特性：
+
+- 全局临时表的表定义会持久化，对所有会话可见
+- 全局临时表的数据只对当前的事务内可见，事务结束后数据自动清空
+- 全局临时表不能与普通表同名
+
+用户可通过 `CREATE GLOBAL TEMPORARY TABLE` 语句创建全局临时表，语句末尾要加上 `ON COMMIT DELETE ROWS`。可通过 `DROP TABLE` 或 `DROP GLOBAL TEMPORARY TABLE` 语句删除全局临时表。
+
+### 全局临时表使用示例
+
```
Comment (TomShawn) [inline temporary-table.md:176]:
```suggestion

> **注意：**
>
> 使用 TiDB 中的临时表前，注意临时表[与其他 TiDB 功能的兼容性限制](#与其他-tidb-功能的兼容性限制)。

```

**Comment by TomShawn** [comment]
LGTM

**Pull request merged**